adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java

```diff
@@ -0,0 +1,265 @@
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.utils.URIBuilder;
+import org.codice.ditto.replication.api.ReplicationException;
+import org.codice.ditto.replication.api.data.CreateStorageRequest;
+import org.codice.ditto.replication.api.data.Metadata;
+import org.codice.ditto.replication.api.data.Resource;
+import org.codice.ditto.replication.api.impl.data.CreateStorageRequestImpl;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WebHdfsClientTest {
+  private static final String HDFS_PORT = "12341";
+  private static final String BASE_URL = "http://localhost:" + HDFS_PORT;
+  private static final Logger LOGGER = LoggerFactory.getLogger(WebHdfsClientTest.class);
+  private static WebHdfsNodeAdapterFactory adapterFactory = new WebHdfsNodeAdapterFactory();
+  private static HdfsLocalCluster hdfsLocalCluster;
+  private static WebHdfsNodeAdapter adapter;
+
+  @BeforeClass
+  public static void setUpClass() throws Exception {
+    LOGGER.info("HDFS instance starting up!");
+    adapter = (WebHdfsNodeAdapter) adapterFactory.create(new URL(BASE_URL));
+    hdfsLocalCluster =
+        new HdfsLocalCluster.Builder()
+            .setHdfsNamenodePort(12345)
+            .setHdfsNamenodeHttpPort(Integer.parseInt(HDFS_PORT))
+            .setHdfsTempDir("embedded_hdfs")
+            .setHdfsNumDatanodes(1)
+            .setHdfsEnablePermissions(false)
+            .setHdfsFormat(true)
+            .setHdfsEnableRunningUserAsProxyUser(true)
+            .setHdfsConfig(new Configuration())
+            .build();
+    hdfsLocalCluster.start();
+  }
+
+  @AfterClass
+  public static void tearDownClass() throws Exception {
+    LOGGER.info("HDFS instance shutting down!");
+    adapter.close();
+    hdfsLocalCluster.stop();
+  }
+
+  @Test
+  public void testDefaultIsAvailable() {
+    assertThat(adapter.isAvailable(), is(true));
+  }
+
+  @Test(expected = ReplicationException.class)
+  public void testInvalidUrlIsNotAvailable() throws MalformedURLException {
+    adapter = (WebHdfsNodeAdapter) adapterFactory.create(new URL("http://foobar:9999"));
+    adapter.isAvailable();
+  }
+
+  @Test(expected = ReplicationException.class)
+  public void testStoppedClusterIsNotAvailable() throws Exception {
+    hdfsLocalCluster.stop();
```

Comment:
I recall you saying that the HDFS cluster was restarting before every test, despite being located in the `@BeforeClass` block, but without that background knowledge, others may be concerned that the cluster isn't being restarted after this test.

Reply:
I can add a comment explaining that behavior.